app/scripts/modules/netflix/fastProperties/scope/fastPropertyScopeSearch.component.ts

```diff
@@ -0,0 +1,223 @@
+          account.regions.forEach((region: any) => acc.add(region.name));
+          return acc;
+        }, new Set());
+
+        this.regions = Array
+          .from(regions)
+          .map((region) => {
+            return {
+              displayName: region,
+              region: region
+            };
+          });
+
+        this.fastPropertyScopeSearchCategoryService.regions = this.regions;
+      })
+      .then(() => {
+        if (this.applicationName && this.applicationName !== 'spinnakerfp') {
+          this.query = this.applicationName;
+          this.executeQuery();
+        }
+      });
+  }
+
+  constructor(
+    private $q: ng.IQService,
+    private infrastructureSearchService: any,
+    private accountService: any,
+    private applicationReader: any,
+    private fastPropertyScopeSearchCategoryService: any
+  ) {
+    this.search = infrastructureSearchService();
+    this.executeQuery = _.debounce(this.executeQuery, 400);
+  }
+
+
+
+  public clearFilters() {
+    this.query = '';
+    this.showSearchResults = false;
+  }
+
+  /*
+   * Select a category item from the list
+   */
+  public selectResult(category: string, selected: any) {
+    this.selectedResult = selected;
+    this.showSearchResults = false;
+    this.scopeOptionsForDisplay = this.fastPropertyScopeSearchCategoryService.buildScopeList(this.categories, category, selected);
+  }
+
+  public displayResults() {
+    this.showSearchResults = true;
+  }
+
+  public toggleNoInpactList(categoryName: string) {
+    this.showNoImpactListForCategory[categoryName] = this.showNoImpactListForCategory[categoryName]
+                                                    ? !this.showNoImpactListForCategory[categoryName]
+                                                    : true;
+
+  }
+
+
+  /*
+   * Query and build the category list
+   */
+  public dispatchQueryInput(evt: any) {
+    this.executeQuery();
+  }
+
+  private executeQuery() {
+    this.querying = true;
+    this.search
+      .query(this.query)
+      .then(this.excludeUnnecessaryCategories)
+      .then(this.filterCategoriesByStartWithQuery)
+      .then(this.addGlobalCategory)
+      .then(this.addRegionCategory)
+      .then(this.fetchApplicationInfo)
+      .then(this.createScopesForEachCategoryResult)
+      .then(this.doneQuerying);
+  }
+
+  /*
+   * Filters
+   */
+  public noImpact(categoryScope: any) {
+    return categoryScope.instanceCounts.up < 1;
+  }
+
+  private excludeUnnecessaryCategories = (results: any[]) => {
+    return this.fastPropertyScopeSearchCategoryService.excludeUnneededCategories(results);
+  };
+
+  private filterCategoriesByStartWithQuery = (categories: any[]): any => {
+    return categories.map((category: any) => {
+      category.results = category.results.filter((r: any) => r.displayName.toLowerCase().startsWith(this.query.toLowerCase()));
+      return category;
+    });
+  };
+
+  public fetchApplicationInfo = (categories: any[]): any  => {
```

Comment:
Category is hard to type here because this data is coming from the global search which returns many different shaped objects.  I perceive that there will be little value or reuse outside of the context o this component.